Strip whitespace in disallow_raw_sql!

gmcgibbon opened the pull request:
### Summary

Closes https://github.com/rails/rails/issues/41307

Support proper SQL parsing for multiple column strings in pluck.

Before:
```
User.pluck("first_name, last_name") #=> SELECT first_name, last_name FROM "users"
User.pluck("first_name, last_name\n") #=> ActiveRecord::UnknownAttributeReference
```

After:
```
User.pluck("first_name, last_name") #=> SELECT first_name, last_name FROM "users"
User.pluck("first_name, last_name\n") #=> SELECT first_name, last_name\n FROM "users"
```

Simply strips whitespace from the SQL sanitization checks of `pluck`.

kaspth commented:
I'd vote for deprecating, since `pluck("first_name, last_name")` won't fit with the Enumerable version of pluck.

gmcgibbon commented:
Hm, this also seems to conflict with invoking SQL functions in pluck. I'll go ahead and try to deprecate instead.

rafaelfranca commented:
`pluck(Arel.sql("some_sql_transformation(first_name)"))` should still be supported. So I don't think not matching Enumerator version of pluck is enough reason to deprecate that.

gmcgibbon committed:
Strip whitespace in disallow_raw_sql!

Removing trailing whitespace when matching columns in
`ActiveRecord::Sanitization.disallow_raw_sql!`.

Co-authored-by: Adrian Hirt <aedu.hirt@gmail.com>

gmcgibbon commented:
Given I can't deprecate string / comma delimited strings, I've decided to simply use `strip` in the SQL sanitizing method. Parsing SQL into an AST isn't worth it with all the edge cases and potential speed regressions.
